File: fibricheck-camera-sdk/src/main/java/com/qompium/fibricheck_camera_sdk/measurement/Yuv.java
package com.qompium.fibricheck_camera_sdk.measurement;

public class Yuv {

  public double y;

  public double u;

  public double v;

  public Yuv (double y, double u, double v) {

    this.y = y;
    this.u = u;
    this.v = v;
  }

  public Yuv (int[] yuvArray, double frameSize) {

    this.y = yuvArray[0] / frameSize;
    this.u = yuvArray[1] / frameSize;
    this.v = yuvArray[2] / frameSize;
  }
}

File: camerasdk/src/main/java/com/qompium/fibricheck/camerasdk/measurement/Yuv.kt
package com.qompium.fibricheck.camerasdk.measurement

data class Yuv(
  var y: Double,
  var u: Double,
  var v: Double
) {
  constructor(yuvArray: IntArray, frameSize: Double) : this(
    yuvArray[0] / frameSize,
    yuvArray[1] / frameSize,
    yuvArray[2] / frameSize
  )

  override fun toString(): String {
    return "{ \"y\": ${y}, \"u\": ${u}, \"v\": ${v} }"
  }
}
